### PR TITLE
chore(agw): Move `make format_all` into a separate bash script

### LINKED ISSUE #14904: Move clang format into shell script
The make target `make format_all` in `lte/gateway/Makefile` should be moved into a shell script. The target does not depend on any other targets and nothing is built. The execution should be easy and convenient. The new command should be communicated and documented.

### PR DESCRIPTION
**Pull request by MoritzThomasHuebner**
Closes #14904 

## Summary

- [x] Moved the make action into a separate bash script.
- [x] Introduced a loop to decrease code redundancy.
- [x] Updated a relevant readme.
- [x] Communicate the change with the community.

## Test Plan

- [x] Executing the script locally fixes formatting errors in the three defined folders.

### CODE REVIEW

File: lte/gateway/format_all.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+# Copyright 2021 The Magma Authors.
```
Comment (nstng) [inline lte/gateway/format_all.sh]: 2023

File: lte/gateway/format_all.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
```
Comment (nstng) [inline lte/gateway/format_all.sh]: couple of notes that are more or less NITs:
* name: this script is only for c code, format_all does not reflect that
* is there a better folder for this script - dev-tools maybe?
* this script fails if clang format is not installed - case on my host, but it's installed in the mgma vm
  * should we have feed-back in this case?
* the script takes some time, but is very silent
  * should we have some start, finish and current folder logging (or even current file)?
* using `set -euo pipefail` is a good pattern

File: lte/gateway/format_all.sh (hunk 1 above)
Comment (MoritzThomasHuebner) [reply lte/gateway/format_all.sh]: :+1:

File: lte/gateway/format_all.sh (hunk 2 above)
Comment (MoritzThomasHuebner) [reply lte/gateway/format_all.sh]: Replying point by point:
- Unsusre, I think we could name it `format_c.sh` perhaps
- I was thinking about that, but I did not find a good option. `dev_tools.py` is file in the directory `format_all.sh` currently is in. The only other `dev_tools` folder is in `cwf`, which would be out of place.
- I added a check if clang is installed and the script now exits on a helpful message.
- I changed it so it prints out every file
- I added the pipefail line

File: dev_tools/clang_format.sh [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script formats all C/C++ files in the magma repo using clang-format-11.
+# The script can be executed on the dev VM or on the
+# host machine if clang-format-11 is installed.
```
Comment (LKreutzer) [inline dev_tools/clang_format.sh:15]: Careful, if on the host `MAGMA_ROOT` won't be defined. It might be best to not support this for now.

Comment (LKreutzer) [reply dev_tools/clang_format.sh:15]: Or you can ask for the user to set the variable correctly.

File: dev_tools/clang_format.sh [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+CLANG_FILE=/usr/bin/clang-format-11
+if test ! -f "$CLANG_FILE"; then
+  echo "clang-format-11 not found. You should execute the script on the dev VM
+    or install clang-format-11 on your host machine."
+  exit 1
+fi
+
+for dir in /orc8r/gateway/c/ /lte/gateway/c/ /lte/gateway/python/;
+do
+  FILES=$(find "${MAGMA_ROOT}/${dir}" \( -iname "*.c" -o -iname "*.cpp" -o -iname "*.h" -o -iname "*.hpp" \) -print)
```
Comment (LKreutzer) [inline dev_tools/clang_format.sh:32]: The slash between `${MAGMA_ROOT}/${dir}` is too much, the `dir` already start with a slash`.

**Review by LKreutzer — CHANGES_REQUESTED** [review]

File: dev_tools/clang_format.sh (hunk 4 above)
Comment (voisey) [reply dev_tools/clang_format.sh:32]: Imo it makes it easier to parse as a path when reading the code though.

Comment (LKreutzer) [reply dev_tools/clang_format.sh:32]: There are two options: either remove the slash here or in the list in line 26 at the beginning of each path. Double slashes work on Linux usually, but I would not want to keep it this way.

File: dev_tools/clang_format.sh (hunk 3 above)
Comment (voisey) [reply dev_tools/clang_format.sh:15]: Regarding "on the host VM", do you mean the host?

Comment (LKreutzer) [reply dev_tools/clang_format.sh:15]: Sorry - of course - edited.

File: dev_tools/clang_format.sh (hunk 4 above)
Comment (voisey) [reply dev_tools/clang_format.sh:32]: Yes, I would favour the latter option.

**Comment by voisey** [comment]
@LKreutzer I assume the reference to `make format_all` should be removed from the table at https://github.com/magma/magma/wiki/Bazel-vs.-Make-commands-dictionary#the-dictionary-for-the-ltegatewaymakefile when this is merged. Is that right?

**Comment by LKreutzer** [comment]
> @LKreutzer I assume the reference to `make format_all` should be removed from the table at https://github.com/magma/magma/wiki/Bazel-vs.-Make-commands-dictionary#the-dictionary-for-the-ltegatewaymakefile when this is merged. Is that right?

I would not remove it, but simply enter the new command on the "Bazel" column, even if it is not strictly a Bazel command.

File: dev_tools/clang_format.sh (hunk 3 above)
Comment (MoritzThomasHuebner) [reply dev_tools/clang_format.sh:15]: Done

File: dev_tools/clang_format.sh (hunk 4 above)
Comment (MoritzThomasHuebner) [reply dev_tools/clang_format.sh:32]: Done, removed the slash in L26

**Comment by MoritzThomasHuebner** [comment]
I made the requested changes. Regarding updating the table, I put this in a follow up ticket #15048, along with a reminder to update the community about the changes.

**Review by voisey — APPROVED** [review]

**Review by LKreutzer — APPROVED** [review]

**Pull request merged**